GoogleForCreators/web-stories-wp
# Dashboard: Update Grid Item UX for links

## Context

Came up recently when updating capabilities in the dashboard that it was inconsistent when a contributor saw a link to 'edit story' when they really can't. This fixes that. 

## Summary

Updates what's clickable to navigate from the dashboard to editing a story. It used to be the story title, now it's just clicking the story itself.

![link vs no link](https://user-images.githubusercontent.com/10720454/130713934-4e9cd6ad-fde3-4fef-8430-bcf112e44693.gif)


## Relevant Technical Choices

- When a user has edit capabilities (`story.capabilities.hasEditAction`)  an anchor element gets added to the entire grid item, positioned absolutely so that it doesn't interfere with the ability to open and interact with the context menu or renaming a story. 
- While here, I found a regression on the focus in the grid that exists in staging as well, thought I'd clean it up. I wanted to make sure I was fixing it in a way that wouldn't resurrect with the update @samwhale had in #8719 so I rebased on that while it was still in review (that's why the CLA is mad). Fixed the issue and cleaned up the unit tests to make sure of it. Now when the context menu is closed, check to see if it was with tab or shift + tab and refocus accordingly.
- I tried to clean up areas that I was already working in. We've got a clean up ticket (#8746 ) that will take care of a good chunk of this. Biggest thing I addressed here was forwarding the ref needed to control the focus of the grid (using arrows to navigate so ya don't get trapped tabbing through endlessly) to the gridItem. 

## To-do

- It'd be nice to separate out the karma tests on the dashboard - the 'my stories' file is so big and daunting, it's easy for things to get lost in there. Separating it out would make it easier to have more robust test sets for different functionality (like maintaining proper focus). 

## User-facing changes

If you have the ability to edit a story, you will get a purple border around grid items in the grid view of 'my stories' when you hover or focus on that item. If you do not have edit capabilities of a story then you will not see this nor will the story be clickable.

## Testing Instructions

Have a few stories in draft and published by at least an admin and contributor. Log in as both, hover and focus on stories. See that when you're a contributor you can only edit contributor stories and the published admin stories are not editable for you. 




This PR can be tested by following these steps:

1.


## Reviews

### Does this PR have a security-related impact?

No

### Does this PR change what data or activity we track or use?

No

### Does this PR have a legal-related impact?

No

## Checklist




---



Fixes #8652
